zebra-state/src/service/finalized_state/tests/vectors.rs

```diff
@@ -0,0 +1,40 @@
+use std::sync::Arc;
+
+use zebra_chain::{
+    block::{Block, Height},
+    parameters::Network,
+    serialization::ZcashDeserializeInto,
+};
+use zebra_test::prelude::*;
+
+use crate::{
+    config::Config,
+    service::finalized_state::{FinalizedBlock, FinalizedState},
+};
+
+use self::assert_eq;
+
+#[test]
+fn blocks_with_v5_transactions() -> Result<()> {
+    zebra_test::init();
+
+    let genesis_block: Arc<Block> =
+        zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES.zcash_deserialize_into()?;
```

Comment:
Instead of deserializing the genesis block in each test, let's modify `PreparedChain` so the first block it generates has the `GENESIS_PREVIOUS_BLOCK_HASH` as its previous block.